examples/autokey_example/main.tf

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "autokey" {
+  source = "../.."
+
+  project_id            = var.project_id
+  keyring               = "example-keyring"
+  location              = var.location
+  keys                  = []
+  autokey_folder_number = var.folder_id
+  autokey_handles = {
+    storage_bucket = {
+      name                   = "bucket-key-handle",
+      project                = var.autokey_resource_project_id,
+      resource_type_selector = "storage.googleapis.com/Bucket",
+      location               = var.location
+    }
+    compute_disk = {
+      name                   = "disk-key-handle",
+      project                = var.autokey_resource_project_id,
+      resource_type_selector = "compute.googleapis.com/Disk",
+      location               = var.location
+    }
+    bigquery_dataset = {
+      name                   = "dataset-key-handle",
+      project                = var.autokey_resource_project_id,
+      resource_type_selector = "bigquery.googleapis.com/Dataset",
```

Comment:
Do we have a test to validate what happens when the resource type selector is malformed or an unsupported type is used?

Reply:
Right now, we don't. Currently, if the user sets an unsupported value, an error during apply will occur.
e.g: `Error creating KeyHandle: googleapi: Error 501: Unknown resource type selector: compute.googleapis.com/Diskw`

It's a possibility to add a variable validator and provide a static list with the allowed values and add a proper test, but I see some problems with that approach:
1. we would need to maintain this allowed values list;
2. even if we update the list quickly when a new service is covered, in order to reflect the change for users a new kms module release will be required - which takes some time;

So, how about we just update the `resource_type_selector` variable description and provide the link for supported services? That way, users can always check it the last info available.
Also, the error message is pretty clear in case of a typo or event if an unsupported type is used.

WDYT?

Reply:
Thanks for investigating.  Yes that will work so users can find the correct values.